frontend/src/components/placementVisuals/Cell.tsx

```diff
@@ -119,38 +168,74 @@ const Cell = ({ container, coordinates, cellSize }: CellProps) => {
+        if (!activeSourceContainer) throw new Error("No active source container when selecting cell color")
+
+        // if no existing sample, no placed sample, and no preview, then it is empty
+        if (placements.length === 0 && cell.getPreview() === null) return EMPTY_CELL_COLOR
+
+        // is being preview?
+        if (cell.getPreview() !== null) {
+            const containerIndex = sourceContainers.findIndex((container) => container.name === activeSourceContainer.name)
+            if (containerIndex === -1)
+                throw new Error(`For preview cell, couldn't find container index for ${activeSourceContainer.name}`)
+            if (containerIndex >= VALID_PREVIEW_CELL_COLORS.length)
+                throw new Error(`Did not expect more than ${VALID_PREVIEW_CELL_COLORS.length} source containers for preview cell color`)
+            return placementState.error ? INVALID_PREVIEW_CELL_COLOR : VALID_PREVIEW_CELL_COLORS[containerIndex]
+        }
+
+        // is selected?
+        const selections = placements.filter((sample) => sample.getSelected())
+        if (selections.length > 0) {
+            return SELECTION_CELL_COLOR
+        }
+
+        const isSource = activeSourceContainer.name === cellID.fromContainer.name
         const existingSample = cell.findExistingSample()
-        const selections = placements.filter((sample) => sample.selected)
-
-        if (selections.length > 0) return "#86ebc1"
-        if (cell.preview !== null) return placementState.error ? "pink" : "#74bbfc"
-        const isSource = cellID.fromContainer.name === sourceContainer?.name
-        if (
-            (isSource && existingSample) ||
-            (!isSource && placements.some((placement) => !existingSample || !placement.sample.sameSampleAs(existingSample)))
-        ) return "#1890ff"
-        if (!isSource && existingSample) return "grey"
-        return "white"
+        if (// source cell with existing sample
+            (isSource && existingSample)
+            ||
+            // destination cell with a placed sample (and not already existing in that cell)
+            (!isSource && placements.some((placement) => !existingSample || !placement.sample.sameSampleAs(existingSample.rawIdentifier())))
+        ) {
+            // assume only one sample per cell. placements also should include existing samples.
+            const sample = placements[0].sample
+            const containerSource = sample.getFromCell()?.getFromContainer()
+            const containerIndex = sourceContainers.findIndex((container) => container.name === (
+                // null is tubes without parent
+                // now that i think about it, cell is never visible for tubes without parent
+                containerSource ? containerSource.getName() : null
+            ))
+            if (containerIndex === -1)
+                throw new Error(`For active cell, couldn't find container index for sample ${containerSource?.getName()}`)
```

Comment:
Don't throw errors for these... Just use a default color ... Not red ... The user did nothing wrong the code should not break here.

Reply:
There is no good reason on the user side to have this fail the samples are in existing containers ... The idea is not bad though ... if this would happen better let someone know ... My fear is that user might not submit an issue for us if it does not fail properly (page not render).